morpholopy/galaxy_plots.py

```diff
@@ -65,12 +66,54 @@
 bingrid = 0.8 * unyt.kpc
 npix_coarse = int(2.0 * r_img_kpc / bingrid)
 
-cmap = plt.get_cmap("Greens")
+cmap = pl.get_cmap("Greens")
 usergreen = cmap(0.7)
-cmap = plt.get_cmap("Blues")
+cmap = pl.get_cmap("Blues")
 userblue = cmap(0.7)
 
 
+def exponential(x, Sigma0, H, offset):
+    return Sigma0 * np.exp(-np.abs(x + offset) / H)
+
+def calculate_scaleheight_pointmasses(zcoords, weights, zrange, resolution):
+
+    S_1D, bin_edges = np.histogram(zcoords, bins = resolution, range = (-zrange, zrange),
+                        weights = weights, density = False)
+
+    z_1D = (bin_edges[1:] + bin_edges[:-1])/2.
+    p0 = (hist.max(), 1., 0.)
+
+    try:
+        popt, pcov = curve_fit(
+                exponential, z_1D[np.isfinite(S_1D)], S_1D[np.isfinite(S_1D)]
+        )
+    except:
+        return np.nan
```

Comment:
Is that the best thing to return if we want to add things to a plot? Wouldn't `-1` or some other unexpected (but "plottable")  value be better?

Reply:
I thought it is safer to have something that does not show up on the plots at all, instead of having maybe the median affected by values such as -1. We can of course check against that by only plotting points with values > 0 but then it is the same again as plotting only values where np.isfinite() - which also catches other problems (inf, nan).

Reply:
Ok. Should we then indicate somewhere that galaxies for which the fits failed are not displayed?

Reply:
In the image caption it already says: "All galaxies with a successful fit are shown." or "Only active galaxies with a successful fit are shown." Is that enough? Or should we make it more clear somewhere?

Reply:
My bad. I missed that. Then all good.